src/sentry/sentry_metrics/indexer/ratelimiters.py

```diff
@@ -107,24 +104,28 @@ def __exit__(self, exc_type: Any, exc_value: Any, traceback: Any) -> None:
         Consumes the rate limits returned by `check_write_limits`.
         """
         if exc_type is None:
-            self._writes_limiter._get_rate_limiter(self._use_case_id).use_quotas(
+            self._writes_limiter._get_rate_limiter(self._use_case_id, self._namespace).use_quotas(
                 self._requests, self._grants, self._timestamp
             )
 
 
 class WritesLimiter:
     def __init__(self) -> None:
-        self.rate_limiters: MutableMapping[UseCaseKey, RedisSlidingWindowRateLimiter] = {}
+        self.rate_limiters: MutableMapping[str, RedisSlidingWindowRateLimiter] = {}
 
-    def _get_rate_limiter(self, use_case_id: UseCaseKey) -> RedisSlidingWindowRateLimiter:
-        if use_case_id not in self.rate_limiters:
+    def _get_rate_limiter(
+        self, use_case_id: UseCaseKey, namespace: str
+    ) -> RedisSlidingWindowRateLimiter:
+        if namespace not in self.rate_limiters:
             options = get_ingest_config(use_case_id).writes_limiter_cluster_options
-            self.rate_limiters[use_case_id] = RedisSlidingWindowRateLimiter(**options)
+            self.rate_limiters[namespace] = RedisSlidingWindowRateLimiter(**options)
```

Comment:
This seems it will break if there is a conflict between namespaces in the configuraiton. It was harder before as the namespace was an enum. 

Please add a unit test that guarantees that certain invariants of the Configuration are always verified (or the test fails and deployment is not possible). Specifically load all the configurations and verify that there is no namespace collision.